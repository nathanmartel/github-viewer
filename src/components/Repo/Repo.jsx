import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ name }) => (
  <>
    <h4>{ name }</h4>
  </>
);

Repo.propTypes = {
  name: PropTypes.string.isRequired
};

export default Repo;
