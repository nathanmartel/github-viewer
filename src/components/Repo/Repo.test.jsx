import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo';

describe('Repo component', () => {
  it('renders Repo', () => {
    const name = 'Lorem Ipsum';
    const wrapper = shallow(<Repo name={name} user={'username'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
