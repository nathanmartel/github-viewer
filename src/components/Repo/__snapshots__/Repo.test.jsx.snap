// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo component renders Repo 1`] = `
<Fragment>
  <h4>
    Lorem Ipsum
  </h4>
  <PullRequests
    pullRequests={
      Array [
        Object {
          "id": 1,
          "number": 5,
          "state": "open",
          "title": "Add this",
          "url": "http://www.hello.com",
        },
        Object {
          "id": 2,
          "number": 6,
          "state": "closed",
          "title": "Subtract that",
          "url": "http://www.hello.com",
        },
      ]
    }
  />
</Fragment>
`;
